src/sentry/snuba/tasks.py

```diff
@@ -218,16 +221,23 @@ def _create_in_snuba(subscription: QuerySubscription) -> str:
             snuba_query,
             subscription.project.organization_id,
         )
-        # TODO: determine whether concatenating query_extra is proper
+        params: ParamsType = {
+            "organization_id": subscription.project.organization_id,
+            "project_id": [subscription.project_id],
+        }
+        skip_time_conditions = True
+        if subscription.timebox_start and subscription.timebox_end:
+            skip_time_conditions = False
+            params["start"] = subscription.timebox_start
+            params["end"] = subscription.timebox_end
+
         snql_query = build_query_builder(
             entity_subscription=entity_subscription,
             query=f'{snuba_query.query}{f" and {subscription.query_extra}" if subscription.query_extra else ""}',
             project_ids=[subscription.project_id],
             environment=snuba_query.environment,
-            params={
-                "organization_id": subscription.project.organization_id,
-                "project_id": [subscription.project_id],
-            },
+            params=params,
+            skip_time_conditions=skip_time_conditions,
         ).get_snql_query()
 
         return _create_snql_in_snuba(subscription, snuba_query, snql_query, entity_subscription)
```

Comment:
The request to Snuba is failing in the tests due to 
`Cannot have existing conditions on time field timestamp`

I've found the error message is coming from
[here](https://github.com/getsentry/snuba/blob/eca97d1c332adabb91d0ba99db33cbbf8f53a3e9/snuba/query/validation/validators.py#L197-L228)
but i'm not entirely sure how it gets here or what we would need to add to resolve this 🤔

Reply:
@wmak Would you have an idea?

cc @ayirr7 @enochtangg ?